stubs/paramiko/paramiko/client.pyi

```diff
@@ -10,6 +10,11 @@ from paramiko.util import ClosingContextManager
 
 from .transport import _SocketLike
 
+class _TransportFactory(Protocol):
+    def __call__(
+        self, sock: _SocketLike, gss_kex: bool, gss_deleg_creds: bool, disabled_algorithms: dict[str, Iterable[str]] | None
```

Comment:
```suggestion
        self, __sock: _SocketLike, *, gss_kex: bool, gss_deleg_creds: bool, disabled_algorithms: dict[str, Iterable[str]] | None
```
This param is passed positionally, the others as keywords.

Reply:
I copied args from `Transport.__init__`. But yes, this one can be pos-only in the callback.